synapse/handlers/federation_event.py

```diff
@@ -2173,28 +2174,28 @@ async def persist_events_and_notify(
             # Note that this returns the events that were persisted, which may not be
             # the same as were passed in if some were deduplicated due to transaction IDs.
             (
-                events,
+                output_events,
                 max_stream_token,
             ) = await self._storage_controllers.persistence.persist_events(
                 event_and_contexts, backfilled=backfilled
             )
 
             if self._ephemeral_messages_enabled:
-                for event in events:
+                for event in output_events:
                     # If there's an expiry timestamp on the event, schedule its expiry.
                     self._message_handler.maybe_schedule_expiry(event)
 
             if not backfilled:  # Never notify for backfilled events
                 with start_active_span("notify_persisted_events"):
                     set_tag(
                         SynapseTags.RESULT_PREFIX + "event_ids",
-                        str([ev.event_id for ev in events]),
+                        str([ev.event_id for ev in output_events]),
                     )
                     set_tag(
                         SynapseTags.RESULT_PREFIX + "event_ids.length",
-                        str(len(events)),
+                        str(len(output_events)),
                     )
-                    for event in events:
+                    for event in output_events:
```

Comment:
Why are these changes necessary?

Reply:
Residue of previous attempt for solution I mostly rolled back. I considered changing this back as well, but it improved readability of which events are we currently talking about for me, so I decided to keep it if it doesn't bother anyone. If so, I will change it back.